src/materials/SteelCreepDamage.C

```diff
@@ -0,0 +1,189 @@
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SteelCreepDamage.h"
+
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+#include "ElasticityTensorTools.h"
+#include "MooseUtils.h"
+#include <cmath>
+#include "RankTwoScalarTools.h"
+
+#include "libmesh/utility.h"
+
+registerMooseObject("BlackBearApp", SteelCreepDamage);
+
+InputParameters
+SteelCreepDamage::validParams()
+{
+  InputParameters params = ScalarDamageBase::validParams();
+  params.addClassDescription("Steel scalar damage model: Material 'suddenly' loses load-carrying "
+                             "capacity. This can model, e.g., 316H creep failure. See "
+                             "'Creep failure simulations of 316H at 550C: Part I - A method and "
+                             "validation, by Oh et al, Engineering Fracture Mechanics 78 (2011)'");
+  params.addRequiredParam<Real>(
+      "epsilon_f",
+      "epsilon_f parameter refers to uniaxial creep fracture strain (creep ductility).");
+  params.addRequiredParam<Real>("creep_law_exponent", "Exponent of the creep power law.");
+  params.addParam<Real>(
+      "reduce_factor",
+      1000.0,
+      "reduce_factor parameter refers to a reduction on the load-carrying capacity (stress).");
+  params.addRangeCheckedParam<Real>("initial_damage_value",
+                                    0.9,
+                                    "initial_damage_value <= 1.0 & initial_damage_value >= 0.0",
+                                    "Initial value of damage that will trigger linear reduction of "
+                                    "quadrature point's load-carrying capacity.");
+  params.addParam<std::string>("creep_strain_name",
+                               "creep_strain",
+                               "Name of the creep strain material driving damage failure.");
+  return params;
+}
+
+SteelCreepDamage::SteelCreepDamage(const InputParameters & parameters)
+  : ScalarDamageBase(parameters),
+    GuaranteeConsumer(this),
+    _creep_strain_name(getParam<std::string>("creep_strain_name")),
+    _creep_strain(getMaterialProperty<RankTwoTensor>(_base_name + _creep_strain_name)),
+    _creep_strain_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + _creep_strain_name)),
+    _epsilon_f(getParam<Real>("epsilon_f")),
+    _reduce_factor(getParam<Real>("reduce_factor")),
+    _initial_damage_value(getParam<Real>("initial_damage_value")),
+    _creep_law_exponent(getParam<Real>("creep_law_exponent")),
+    _stress(getMaterialProperty<RankTwoTensor>(_base_name + "stress"))
+{
+}
```

Comment:
add a check `_creep_law_exponent != -0.5` (you never know what ideas users might get...)